packages/react-search-ui/src/SearchProvider.js

```diff
@@ -42,16 +41,16 @@ class SearchProvider extends Component {
 
   render() {
     const { children } = this.props;
-    if (!this.driver) return null;
+
+    if (!this.state.driver) return null;
 
     const providerValue = {
-      ...this.state,
-      ...this.driver.getActions()
+      driver: this.state.driver
     };
 
     return (
       <SearchContext.Provider value={providerValue}>
```

Comment:
This line equivalent to `<SearchContext.Provider value={{ driver: this.state.driver }}>`. 

**From the react doc**
> Because context uses reference identity to determine when to re-render, there are some gotchas that could trigger unintentional renders in consumers when a provider’s parent re-renders. For example, the code below will re-render all consumers every time the Provider re-renders because a new object is always created for value.

https://reactjs.org/docs/context.html#caveats

**More information about this**
* https://medium.com/@ryanflorence/react-context-and-re-renders-react-take-the-wheel-cd1d20663647
* https://kentcdodds.com/blog/how-to-optimize-your-context-value

**Solution 1**
```js
// pseudocode

class SearchProvider extends Component {
  constructor(props) {
    super(props);
    this.state = {
      driver: new SearchDriver(props.config)
    };
  }

  componentWillUnmount() {
    this.state.driver.tearDown();
  }

  render() {
    if (!this.state.driver) return null; // do we need this line?

   return (
      <SearchContext.Provider value={this.state.driver}>
         {this.props.children}
      </SearchContext.Provider>
   );
  }
}
```
**Solution 2 with hooks**
```js
import React, { useState, useEffect } from 'react';

function SearchProvider(props) {
  const [driver,] = useState(() => new SearchDriver(props.config));

  useEffect(() => driver.tearDown, []);

  if (!driver) return null; // Do we need this line?
  
  return <SearchContext.Provider value={driver} {...props} />
}
```
Let me know what do you think